packages/jaeger-ui/src/utils/helpers/ReactShallowRenderer.js

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 The Jaeger Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable import/no-extraneous-dependencies */
+import React from 'react';
+import ShallowRenderer from 'react-test-renderer/shallow';
+import { isFragment, isLazy, isPortal, isMemo, isSuspense, isForwardRef } from 'react-is';
+
+const shallow = (element, options) => {
+  const localOptions = options === undefined ? {} : options;
```

Comment:
what is the origin of this code?

Reply:
I mean the whole file?

Reply:
I got it from a [blog here ](https://thesametech.com/snapshot-testing-in-rtl/#:~:text=import%20React%20from,transformNode(this.shallowWrapper)%3B%0A%20%20%7D%0A%7D)
 which I have further modified to fix the lint errors, and to directly call the shallow function instead of using `new ReactShallowRenderer`